components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSessionState.kt

```diff
@@ -56,19 +57,32 @@ class GeckoEngineSessionState internal constructor(
             reader.beginObject()
 
             val key = reader.nextName()
-            if (key != GECKO_STATE_KEY) {
-                throw AssertionError("Unknown state key: $key")
-            }
 
-            val rawState = reader.nextString()
+            val rawState = if (key == GECKO_STATE_KEY) {
+                reader.nextString()
+            } else {
+                reader.consumeObject()
+                null
+            }
 
             reader.endObject()
 
-            GeckoEngineSessionState(
-                GeckoSession.SessionState.fromString(rawState)
-            )
+            if (rawState != null) {
+                GeckoEngineSessionState(GeckoSession.SessionState.fromString(rawState))
+            } else {
+                GeckoEngineSessionState(null)
+            }
         } catch (e: IOException) {
             GeckoEngineSessionState(null)
         }
     }
 }
+
+private fun JsonReader.consumeObject() {
+    while (true) {
+        when (peek()) {
```

Comment:
Hm looks like `peek` can also throw an `AssertionError` and `IllegalStateException`, not just the documented `IOException`. Maybe above we should catch all?